src/client/tenants.rs

```diff
@@ -79,6 +79,21 @@ impl Client {
         Ok(res)
     }
 
+    /// Get a tenant by ID.
+    pub async fn get_tenant(&self, id: Uuid) -> Result<Tenant, Error> {
+        let req = self.build_request(Method::GET, TENANT_PATH.chain_one(id));
+        let mut res: Vec<Tenant> = self.send_request(req).await?;
+        match res.len() {
+            0 => Err(Error::Api(error::ApiError {
+                status_code: StatusCode::NOT_FOUND,
+                messages: vec!["Tenant not found".to_string()],
+            })),
+            // Fine to assume only one item is returned here as Frontegg provides a uniqueness
+            // guarantee
+            _ => Ok(res.remove(0)),
+        }
```

Comment:
100% optional suggestion: you could use an API like pop, if you'd like to avoid the panic-on-out-of-bounds nature of `remove`.

```suggestion
        match res.pop() {
            None => Err(Error::Api(error::ApiError {
                status_code: StatusCode::NOT_FOUND,
                messages: vec!["Tenant not found".to_string()],
            })),
            // Fine to assume only one item is returned here as Frontegg provides a uniqueness
            // guarantee
            Some(tenant) => Ok(tenant),
        }
```

Reply:
Nice, I like it! I updated this to use `pop().ok_or()`.